Samples/Samples/ViewModel/CalendarEventAddViewModel.cs

```diff
@@ -152,6 +236,267 @@ public TimeSpan EndTime
+                        RecurrenceDays = recurrenceWeekdays;
+                    }
+                    else
+                    {
+                        RecurrenceInterval = 1;
+                    }
+                }
+            }
+        }
+
+        bool isMonthDaySpecific = true;
+
+        public bool IsMonthDaySpecific
+        {
+            get => isMonthDaySpecific;
+            set
+            {
+                if (SetProperty(ref isMonthDaySpecific, value))
+                {
+                    if (value)
+                    {
+                        SelectedRecurrenceMonthDay = 1;
+                        SelectedRecurrenceMonthWeek = string.Empty;
+                        SelectedMonthWeekRecurrenceDay = string.Empty;
+                    }
+                    else
+                    {
+                        SelectedRecurrenceMonthDay = 0;
+                        SelectedRecurrenceMonthWeek = "First";
+                        SelectedMonthWeekRecurrenceDay = DayOfTheWeek.Monday.ToString();
+                    }
+                }
+            }
+        }
+
+        public ObservableCollection<DayOfTheWeekSwitch> RecurrenceDays { get; set; } = new ObservableCollection<DayOfTheWeekSwitch>()
+        {
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Monday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Tuesday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Wednesday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Thursday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Friday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Saturday },
+            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Sunday }
+        };
+
+        public ObservableCollection<string> RecurrenceTypes { get; } = new ObservableCollection<string>()
+        {
+            RecurrenceFrequency.None.ToString(),
+            RecurrenceFrequency.Daily.ToString(),
+            RecurrenceFrequency.Weekly.ToString(),
+            RecurrenceFrequency.Monthly.ToString(),
+            RecurrenceFrequency.Yearly.ToString()
+        };
+
+        string selectedRecurrenceType = RecurrenceFrequency.None.ToString();
+
+        public string SelectedRecurrenceType
+        {
+            get => selectedRecurrenceType;
+
+            set
+            {
+                if (SetProperty(ref selectedRecurrenceType, value) && selectedRecurrenceType != null)
+                {
+                    OnPropertyChanged(nameof(CanAlterRecurrence));
+                    OnPropertyChanged(nameof(DisplayTimeInformation));
+                    OnPropertyChanged(nameof(IsDaily));
+                    OnPropertyChanged(nameof(IsWeekly));
+                    OnPropertyChanged(nameof(IsMonthlyOrYearly));
+                    OnPropertyChanged(nameof(IsYearly));
+                    OnPropertyChanged(nameof(SelectedRecurrenceTypeDisplay));
+                    RecurrenceInterval = 1;
```

Comment:
Why is this being set here? If you're populating an existing recurrence pattern you probably don't want this set

Reply:
Should be getting handled properly behind the scenes now, it was a fix to some buggy behaviour that was occurring previously